streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -179,11 +181,12 @@ private void close(final boolean clean) {
                 transitionTo(State.CLOSING);
             }
 
-            if (state() == State.CLOSING) {
-                executeAndMaybeSwallow(clean, () -> {
-                    StateManagerUtil.closeStateManager(log, logPrefix, clean,
-                        false, stateMgr, stateDirectory, TaskType.STANDBY);
-                }, "state manager close", log);
+            if (state() == State.CLOSING || state() == State.RESTORING) {
+                // see StreamTask.close() for the explanation
+                final boolean wipeStateStore = !clean && !eosDisabled;
```

Comment:
fix 2

Reply:
I think this is what the below TODO (191) was added for, Thanks :) Please feel free to remove that TODO marker then.